### PR TITLE
fix: GOG cloud save fetch failure handling

### PR DESCRIPTION
**Pull request by kiequoo**
## Description
This is a split-out PR from #1094 focused on the GOG cloud save fetch and timestamp fixes.

Fixes two bugs in GOG cloud save sync and one related issue in save-location resolution:

- `getCloudFiles()` was treating request or parse failures the same as a valid empty cloud result, which could make sync logic believe the cloud was empty and incorrectly choose an upload path.
- GOG `last_modified` timestamps were parsed with `Instant.parse(...)`, which does not handle the offset-based format GOG returns — so remote timestamps could be dropped during sync decisions.
- Removes the fake default GOG cloud-save path fallback when the API returns no save locations. That fallback is not useful because in the same case there is also no `clientSecret`, so cloud auth cannot succeed and the path can never be used for a real sync.

**Changes:**
- Return `null` from GOG cloud file listing on HTTP or parse failure instead of `emptyList()`
- Abort sync when cloud file listing fails instead of treating the failure as "cloud is empty"
- Parse GOG cloud timestamps with `OffsetDateTime.parse(...).toInstant()`
- Stop falling back to a fake default cloud-save location when the API reports no save locations
- Add regression tests covering offset timestamp parsing and failure-vs-empty cloud responses

## Checklist
- [x] If I have access to `#code-changes`, I have discussed this change there and it has been green-lighted. If I do not have access, I have still provided clear context in this PR. If I skip both, I accept that this change may face delays in review, may not be reviewed at all, or may be closed.
- [ ] I have attached a recording of the change.
- [x] I have read and agree to the contribution guidelines in `CONTRIBUTING.md`.

### CODE REVIEW

**Comment by phobos665** [comment]
Please make sure to use the PR template :)

File: app/src/main/java/app/gamenative/service/gog/GOGCloudSavesManager.kt [hunk 1]
```diff
@@ -434,10 +408,56 @@ class GOGCloudSavesManager(
...
     }
 
+    internal fun parseCloudTimestamp(lastModified: String): Long? =
+        try {
+            // GOG returns timestamps like "2026-04-02T20:34:00.123456+00:00".
+            OffsetDateTime.parse(lastModified).toInstant().epochSecond
+        } catch (_: DateTimeParseException) {
+            null
+        }
+
```
Comment (phobos665) [inline app/src/main/java/app/gamenative/service/gog/GOGCloudSavesManager.kt:460]: Nice, I like these being split out into internal files for easy unit testing
